aten/src/ATen/native/quantized/README.md

```diff
@@ -0,0 +1,213 @@
+The quantized folder holds the implementation of the low-level quantized kernel.
+The kernels are registered in `torch::_ops` namespace, and operate on the quantized `at::Tensor` data type.
+
+This document serves as an entry point for quantized kernel implementation.
+
+## Implementing native quantized ops
+
+The new quantized ops are almost always located under the `ATen/native/quantized/cpu` folder.
+For the sake of an example, let us implement an element-wise quantized logical AND operation under `ATen/native/quantized/cpu/qand.cpp`.
+
+### Step 0. Implement the quantized function
+
+Before writing the quantized kernel and registering it, let us implement a quantized function.
+That would assist in any further discussion.
+The snippet below shows the implementation of a quantized AND operator, with the support of all implemented quantized types.
+
+```c++
+Tensor quantized_and(Tensor qa, Tensor qb) {
+  // Some type checks for qa and qb should be here...
+  Tensor qc;
+  double scale = qa.q_scale();
+  long zero_point qa.q_zero_point();
+
+  auto iter = TensorIterator::binary_op(qc, qa, qb);
+
+  AT_DISPATCH_QINT_TYPES(qa.scalar_type(), "quantized_and", [&]() {
+    Tensor qc = at::_empty_affine_quantized(qa.sizes(),
```

Comment:
technically it can be pulled out of the switch, can't it?

Reply:
I don't think it can -- it uses `at::device(kCPU).dtype(SCALAR_TYPE)` as the next argument, and `SCALAR_TYPE` is available only within the switch